modules/pricer/src/test/java/com/opengamma/strata/pricer/calibration/CalibrationZeroRateUsdEur2OisFxTest.java

```diff
@@ -298,10 +301,10 @@ private void calibration_market_quote_sensitivity_check(
     }
     double[] mqsEur2Computed = mqs.getSensitivity(EUR_DSC_CURVE_NAME, EUR).getSensitivity().toArray();
     for (int i = 0; i < EUR_DSC_NB_NODES; i++) {
-      Map<ObservableKey, Double> map = new HashMap<>(ALL_QUOTES.getValues());
+      Map<MarketDataKey<?>, Object> map = new HashMap<>(ALL_QUOTES.getValues());
       map.put(QuoteKey.of(StandardId.of(SCHEME, EUR_DSC_ID_VALUE[i])), EUR_DSC_MARKET_QUOTES[i] + shift);
-      ImmutableObservableValues ov = ImmutableObservableValues.of(map);
-      ImmutableRatesProvider rpShifted = calibrator.apply(ov);
+      ImmutableMarketData marketData = ImmutableMarketData.builder().values(map).build();
```

Comment:
Looking at al the use cases that needed changing, adding a `MarketData.of(Map)` method looks worthwhile.